modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_0_0/DDMStructureUpgradeProcess.java

```diff
@@ -166,8 +169,19 @@ private void _upgradeStructureLayoutDefinition() throws Exception {
 
 					preparedStatement2.setLong(
 						2, resultSet.getLong("classNameId"));
-					preparedStatement2.setString(
-						3, resultSet.getString("structureKey"));
+
+					String structureLayoutKey = String.valueOf(increment());
```

Comment:
Do we need to increment even if the number is not used?

Reply:
Hi @rodrigopaulino , I'm not sure if I understood the question correctly. But I'm using this 'increment' only because I need to update with some value if it's not the default version, since the `structureLayoutkey` can't be empty.

Reply:
No problem, @carolmariaabb. Sorry, I should have been clearer. Your logic is correct about using the `structureLayoutKey`. My question was related to the Counter. The code's always incrementing the Counter table even if it doesn't end up using the generated number.
I think it would be best if we reversed the logic so it only increments if the version is different the default. Does it make sense now? 😄

Reply:
Got it! It makes totally sense now 😅 Thanks a lot, @rodrigopaulino